sdk/Sdk.Generators/FunctionMetadataProviderGenerator/KnownFunctionMetadataTypes.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.Azure.Functions.Worker.Sdk.Generators
+{
+    internal readonly struct KnownFunctionMetadataTypes
+    {
+        internal readonly INamedTypeSymbol? BindingAttribute;
+        internal readonly INamedTypeSymbol? OutputBindingAttribute;
+        internal readonly INamedTypeSymbol? FunctionName;
+        internal readonly INamedTypeSymbol? BindingPropertyNameAttribute;
+        internal readonly INamedTypeSymbol? DefaultValue;
+        internal readonly INamedTypeSymbol? HttpResponse;
+        internal readonly INamedTypeSymbol? HttpTriggerBinding;
```

Comment:
Nullable since customer may not be using http extension package. The 5 non-Http types above should be present in any functions app (as part of abstractions package) and we could throw an error when initializing if it's `null` and make these props non nullable. Would have to add error handling, but as is, these props are used in Symbol comparisons in `FunctionMetadataProviderGenerator.Parser` and returning `null` is still fine.